include/BaseGameState.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <SDL.h>
+
+class GameState
+{
+public:
+	GameState() {}
```

Comment:
Perhaps both of these special member functions (constructor and destructor) can be marked `default` to be more consistent with the rest of the other class' special member functions.